### PR TITLE
chore(docs): Add expand animation to headings

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change an animation to the heading on the main docs page, making the text expand outward as it's being rendered.

Kapture 2022-12-15 at 13 07 02

#### Description of how you validated changes

Manual test in Chrome, Safari and firefox.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/styles/docs/base.scss [hunk 1]
```diff
@@ -303,6 +303,31 @@ h4:hover > a .icon-link {
...
+      animation: expand-out calc(2 * var(--amplify-time-long))
+        cubic-bezier(0.2, 0.6, 0.3, 1) both;
+    }
+  }
+}
+
+@keyframes expand-out {
+  0% {
+    letter-spacing: calc(-1 * var(--amplify-space-xs));
+    transform: translateZ(-500px);
```
Comment (ioanabrooks) [inline docs/src/styles/docs/base.scss:319]: Arbitrary number based on manual testing, it works without this entirely, but the expansion is less smooth and less noticeable.

File: docs/src/components/home/sections/LiveSection.tsx [hunk 2]
```diff
@@ -34,7 +34,12 @@ export const LiveSection = ({ platform, ...rest }) => {
       ref={ref}
     >
       <View className="docs-home-subsection">
-        <Heading level={2}>Take it for a test drive</Heading>
+        <Heading
+          level={2}
+          className={classNames('expand-out', isVisible && 'shown')}
```
[the remark this answers is not in the document]
Comment (ioanabrooks) [reply docs/src/components/home/sections/LiveSection.tsx:39]: Yeah that would be nice, the parent View also has similar logic, but I would make that a separate PR

File: docs/src/styles/docs/base.scss [hunk 3]
```diff
@@ -303,6 +303,31 @@ h4:hover > a .icon-link {
   }
 }
 
+.expand-out {
+  // Only expand things for people without reduced motion preference
+  @media (prefers-reduced-motion: no-preference) {
+    &.shown {
+      animation: expand-out calc(2 * var(--amplify-time-long))
+        cubic-bezier(0.2, 0.6, 0.3, 1) both;
```
Comment (dbanksdesign) [inline docs/src/styles/docs/base.scss:311]: Out of scope for this, but should we add a longer time to our design tokens and should we add easing curves to our design tokens too?

File: docs/src/styles/docs/base.scss (hunk 1 above)
Comment (dbanksdesign) [reply docs/src/styles/docs/base.scss:319]: Could you add a comment in here so a future developer knows this?

**Review by dbanksdesign — APPROVED** [review]
Love it! One minor thing and some thoughts for a future PR

File: docs/src/styles/docs/base.scss (hunk 1 above)
Comment (ioanabrooks) [inline docs/src/styles/docs/base.scss:319]:
```suggestion
    // arbitrary number found through heuristic methods, without this transform the expansion is less smooth and less noticeable.
    transform: translateZ(-500px);
```

File: docs/src/styles/docs/base.scss (hunk 3 above)
Comment (ioanabrooks) [reply docs/src/styles/docs/base.scss:311]: Addingmore animation timings makes perfect sense, for the curves, it's quite subjective and depends on the animation type. Like these particular values work well for this animation but might not for another if that makes sense.

File: docs/src/styles/docs/base.scss (hunk 1 above)
Comment (ioanabrooks) [reply docs/src/styles/docs/base.scss:320]: Done!

**Review by calebpollman — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
🤩

**Pull request merged**